processing/src/main/java/org/apache/druid/frame/read/columnar/StringFrameColumnReader.java

```diff
@@ -60,16 +61,25 @@
 
 import javax.annotation.Nullable;
 import java.nio.ByteBuffer;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 
+/**
+ * Reader for {@link StringFrameColumnWriter}, types {@link ColumnType#STRING} and {@link ColumnType#STRING_ARRAY}.
```

Comment:
at first glance it does seem kind of funny that strings and string arrays live in the same reader, but makes sense because of mvds. Any idea what other array frame columns might look like? I imagine most array columns would look pretty similar, it might be nice to delegate some way to read elements generically so we could just have a shared ArrayFrameColumnReader

Reply:
I am aiming to tackle those issues in a future PR

Reply:
I think the numeric array row-based format will look quite similar to string arrays, except the `VALUE_TERMINATOR` isn't needed so we should skip that. The rest of it will probably be the same. I'm going to see if any code can be shared.

The columnar format will also be similar to the string columnar format, except the length section isn't needed, because the values are all the same length. Same story, we'll skip that and see if any code can be shared.